coreblocks/lsu/pma.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import dataclass
+from amaranth import *
+
+from coreblocks.params import *
+from transactron.utils import HasElaborate
+from transactron.core import TModule
+
+
+@dataclass
+class PMARegion:
+    """
+    Data class for physical memory attributes contiguous region of memory. Region of memory
+    includes both start end end address.
+
+    Attributes
+    ----------
+    start : int
+        Defines beginning of region.
+    end : int
+        Defines end of region.
+    attrs : Record
+        Attributes for given region. Record should follow adequate layout from `PMALayouts`.
+    """
+
+    start: int
+    end: int
+    mmio: bool
+
+
+class PMAChecker(Elaboratable):
+
+    """
+    Implementation of physical memory attributes checker. It may or may not be a part of LSU.
+    `PMAChecker` exposes method `ask` that can be used to request information about given memory
+    address. Result of `ask` method contains bit vector of all defined PMAs.
+
+    Attributes
+    ----------
+
+    addr : Signal
+        Memory address, for which PMAs are requested.
+
+    result : Record
+        PMAs for given address.
+
+    """
```

Comment:
There are some superfluous newlines here.

```suggestion
    """
    Implementation of physical memory attributes checker. It may or may not be a part of LSU.
    `PMAChecker` exposes method `ask` that can be used to request information about given memory
    address. Result of `ask` method contains bit vector of all defined PMAs.

    Attributes
    ----------
    addr : Signal
        Memory address, for which PMAs are requested.
    result : Record
        PMAs for given address.
    """
```